upgrade/change_v21.03.0.go

```diff
@@ -0,0 +1,449 @@
+}
+
+func dropDeprecated(dg *dgo.Dgraph) error {
+	if !Upgrade.Conf.GetBool(deleteOld) {
+		return nil
+	}
+	for pred := range deprecatedPreds {
+		op := &api.Operation{
+			DropOp:    api.Operation_ATTR,
+			DropValue: pred,
+		}
+		if err := alterWithClient(dg, op); err != nil {
+			return errors.Wrapf(err, "error deleting old predicate %s", pred)
+		}
+	}
+	for typ := range deprecatedTypes {
+		op := &api.Operation{
+			DropOp:    api.Operation_TYPE,
+			DropValue: typ,
+		}
+		if err := alterWithClient(dg, op); err != nil {
+			return errors.Wrapf(err, "error deleting old type %s", typ)
+		}
+	}
+	fmt.Println("Successfully dropped the deprecated predicates")
+	return nil
+}
+
+func upgradeCORS() error {
+	dg, cb := x.GetDgraphClient(Upgrade.Conf, true)
+	defer cb()
+
+	jwt, err := getAccessJwt()
+	if err != nil {
+		return errors.Wrap(err, "while getting jwt auth token")
+	}
+
+	// Get CORS.
+	corsData := make(map[string][]cors)
+	if err = getQueryResult(dg, queryCORS_v21_03_0, &corsData); err != nil {
+		return errors.Wrap(err, "error querying cors")
+	}
+
+	var corsList []string
+	var maxUid uint64
+	for _, cors := range corsData["cors"] {
+		uid, err := strconv.ParseUint(cors.UID, 0, 64)
+		if err != nil {
+			return err
+		}
+		if uid > maxUid {
+			maxUid = uid
+			if len(cors.Cors) == 1 && cors.Cors[0] == "*" {
+				// No need to update the GraphQL schema if all origins are allowed.
+				corsList = corsList[:0]
+				continue
+			}
+			corsList = cors.Cors
+		}
+	}
+
+	// Get GraphQL schema.
+	schemaData := make(map[string][]sch)
+	if err = getQueryResult(dg, querySchema_v21_03_0, &schemaData); err != nil {
+		return errors.Wrap(err, "error querying graphql schema")
+	}
+
+	var gqlSchema string
+	maxUid = 0
+	for _, schema := range schemaData["schema"] {
+		uid, err := strconv.ParseUint(schema.UID, 0, 64)
+		if err != nil {
+			return err
+		}
+		if uid > maxUid {
+			maxUid = uid
+			gqlSchema = schema.GQLSchema
+		}
+	}
+
+	// Update the GraphQL schema.
+	if err := updateGQLSchema(jwt, gqlSchema, corsList); err != nil {
+		return err
+	}
+
+	// Drop all the deprecated predicates and types.
+	return dropDeprecated(dg)
+}
+
+func getData(db *badger.DB, attr string, fn func(item *badger.Item) error) error {
+	return db.View(func(txn *badger.Txn) error {
+		attr = x.GalaxyAttr(attr)
+		initKey := x.ParsedKey{
+			Attr: attr,
+		}
+		prefix := initKey.DataPrefix()
+		startKey := append(x.DataKey(attr, math.MaxUint64))
+
+		itOpt := badger.DefaultIteratorOptions
+		itOpt.AllVersions = true
+		itOpt.Reverse = true
+		itOpt.Prefix = prefix
+		itr := txn.NewIterator(itOpt)
+		defer itr.Close()
+		for itr.Seek(startKey); itr.Valid(); itr.Next() {
+			item := itr.Item()
+			// We expect only complete posting list.
+			x.AssertTrue(item.UserMeta() == posting.BitCompletePosting)
+			if err := fn(item); err != nil {
+				return err
+			}
+			break
+		}
+		return nil
+	})
+}
+
+func updateGqlSchema(db *badger.DB, cors [][]byte) error {
```

Comment:
so much confusion. updateGqlSchema(offline)/updateGQLSchema(Online). Can we make this bit clear?

Reply:
Renamed it to include offline/online. Thanks.